src/main/java/teammates/ui/webapi/output/InstructorData.java

```diff
@@ -0,0 +1,20 @@
+package teammates.ui.webapi.output;
+
+import java.util.Optional;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+
+/**
+ * The API output format of {@link InstructorAttributes}.
+ */
+public class InstructorData extends ApiOutput {
+    private final String name;
+
+    public InstructorData(InstructorAttributes instructorAttributes) {
+        this.name = Optional.ofNullable(instructorAttributes.getName()).orElse("DEFAULT_NAME");
```

Comment:
This usage of `Optional` is not correct. This is because `instructorAttributes.getName()` is resolved first, which will cause an NPE to be thrown and render your `Optional` useless.

The correct usage is `Optional.ofNullable(instructorAttributes).map(instructorAttributes::getName).orElse("....")` (something like that).

https://stackoverflow.com/questions/35337020/optional-ofnullable-and-method-chaining

Reply:
(see the other comment about the necessity of this; I just wanted to point out that this usage is not correct)

Reply:
Yup your right